File: WorldFeed/Client/src/app/core/Core/DTO/MessageDataDTO.ts
import {IMessageEntities} from '../../Public/Models/Entities/IMessageEntities';
import {IMessageDataDTO} from "../../Public/Models/Interfaces/DTO/IMessageDataDTO";
import {IQuickReplyDTO} from "../../Public/Models/Interfaces/DTO/IQuickReplyDTO";
import {IQuickReplyResponse} from "../../Public/Models/Interfaces/IQuickReplyResponse";
import {IAttachmentDTO} from "../../Public/Models/Interfaces/DTO/IAttachmentDTO";

export class MessageDataDTO implements IMessageDataDTO {
  // [JsonProperty("text")]
  public text: string;

  // [JsonProperty("entities")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public entities: IMessageEntities;

  // [JsonProperty("quick_reply")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public quickReply: IQuickReplyDTO;

  // [JsonProperty("quick_reply_response")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public quickReplyResponse: IQuickReplyResponse;

  // [JsonProperty("attachment")]
  // [JsonConverter(typeof(JsonPropertyConverterRepository))]
  public attachment: IAttachmentDTO;
}

File: Sharebook/Server/src/Services/Message/Sharebook.Message.API/DTO/MessageDataDTO.cs
namespace Sharebook.Message.DTO
{
    using Newtonsoft.Json;

    using Sharebook.Common.JsonConverters;
    using Sharebook.Common.Public.Models.Entities;
    using Sharebook.Common.Public.Models.Interfaces;
    using Sharebook.Common.Public.Models.Interfaces.DTO;

    public class MessageDataDTO : IMessageDataDTO
    {
        [JsonProperty("text")]
        public string Text { get; set; }

        [JsonProperty("entities")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IMessageEntities Entities { get; set; }

        [JsonProperty("quick_reply")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IQuickReplyDTO QuickReply { get; set; }

        [JsonProperty("quick_reply_response")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IQuickReplyResponse QuickReplyResponse { get; set; }

        [JsonProperty("attachment")]
        [JsonConverter(typeof(JsonPropertyConverterRepository))]
        public IAttachmentDTO Attachment { get; set; }
    }
}
